File: app/src/main/java/com/irishinterest/irishinterest/model/authors/Author.java
package com.irishinterest.irishinterest.model.authors;

import android.os.Build;

import androidx.annotation.RequiresApi;

import com.irishinterest.irishinterest.network.api.helper.Response;
import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "id",
        "firstname",
        "lastname",
        "dob",
        "profile",
        "contact",
        "address",
        "url",
        "image",
        "createdby",
        "altlink"
})
public class Author extends Response {
    @JsonProperty("id")
    private Integer id;
    @JsonProperty("firstname")
    private String firstname;
    @JsonProperty("lastname")
    private String lastname;
    @JsonProperty("dob")
    private String dob;
    @JsonProperty("profile")
    private String profile;
    @JsonProperty("contact")
    private String contact;
    @JsonProperty("address")
    private String address;
    @JsonProperty("url")
    private String url;
    @JsonProperty("image")
    private Object image;
    @JsonProperty("createdby")
    private String createdby;
    @JsonProperty("altlink")
    private String altlink;
    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    @JsonProperty("id")
    public Integer getId() {
        return id;
    }

    @JsonProperty("id")
    public void setId(Integer id) {
        this.id = id;
    }

    @JsonProperty("firstname")
    public String getFirstname() {
        return firstname;
    }

    @JsonProperty("firstname")
    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    @JsonProperty("lastname")
    public String getLastname() {
        return lastname;
    }

    @JsonProperty("lastname")
    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    @JsonProperty("dob")
    public String getDob() {
        return dob;
    }

    @JsonProperty("dob")
    public void setDob(String dob) {
        this.dob = dob;
    }

    @JsonProperty("profile")
    public String getProfile() {
        return profile;
    }

    @JsonProperty("profile")
    public void setProfile(String profile) {
        this.profile = profile;
    }

    @JsonProperty("contact")
    public String getContact() {
        return contact;
    }

    @JsonProperty("contact")
    public void setContact(String contact) {
        this.contact = contact;
    }

    @JsonProperty("address")
    public String getAddress() {
        return address;
    }

    @JsonProperty("address")
    public void setAddress(String address) {
        this.address = address;
    }

    @JsonProperty("url")
    public String getUrl() {
        return url;
    }

    @JsonProperty("url")
    public void setUrl(String url) {
        this.url = url;
    }

    @JsonProperty("image")
    public Object getImage() {
        return image;
    }

    @JsonProperty("image")
    public void setImage(Object image) {
        this.image = image;
    }

    @JsonProperty("createdby")
    public String getCreatedby() {
        return createdby;
    }

    @JsonProperty("createdby")
    public void setCreatedby(String createdby) {
        this.createdby = createdby;
    }

    @JsonProperty("altlink")
    public String getAltlink() {
        return altlink;
    }

    @JsonProperty("altlink")
    public void setAltlink(String altlink) {
        this.altlink = altlink;
    }

    @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }


    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Author)) return false;
        Author author = (Author) o;
        return Objects.equals(id, author.id) &&
                Objects.equals(firstname, author.firstname) &&
                Objects.equals(lastname, author.lastname) &&
                Objects.equals(dob, author.dob) &&
                Objects.equals(profile, author.profile) &&
                Objects.equals(contact, author.contact) &&
                Objects.equals(address, author.address) &&
                Objects.equals(url, author.url) &&
                Objects.equals(image, author.image) &&
                Objects.equals(createdby, author.createdby) &&
                Objects.equals(altlink, author.altlink) &&
                Objects.equals(additionalProperties, author.additionalProperties);
    }

    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
    @Override
    public int hashCode() {
        return Objects.hash(id, firstname, lastname, dob, profile, contact, address, url, image, createdby, altlink, additionalProperties);
    }
}

File: app/src/main/java/com/irishinterest/irishinterest/model/authors/Author.kt
package com.irishinterest.irishinterest.model.authors

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonAnyGetter
import com.fasterxml.jackson.annotation.JsonAnySetter
import androidx.annotation.RequiresApi
import android.os.Build
import com.irishinterest.irishinterest.network.api.helper.Response
import java.util.*

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder(
    "id",
    "firstname",
    "lastname",
    "dob",
    "profile",
    "contact",
    "address",
    "url",
    "image",
    "createdby",
    "altlink"
)
class Author : Response() {
    @get:JsonProperty("id")
    @set:JsonProperty("id")
    @JsonProperty("id")
    var id: Int? = null

    @get:JsonProperty("firstname")
    @set:JsonProperty("firstname")
    @JsonProperty("firstname")
    var firstname: String? = null

    @get:JsonProperty("lastname")
    @set:JsonProperty("lastname")
    @JsonProperty("lastname")
    var lastname: String? = null

    @get:JsonProperty("dob")
    @set:JsonProperty("dob")
    @JsonProperty("dob")
    var dob: String? = null

    @get:JsonProperty("profile")
    @set:JsonProperty("profile")
    @JsonProperty("profile")
    var profile: String? = null

    @get:JsonProperty("contact")
    @set:JsonProperty("contact")
    @JsonProperty("contact")
    var contact: String? = null

    @get:JsonProperty("address")
    @set:JsonProperty("address")
    @JsonProperty("address")
    var address: String? = null

    @get:JsonProperty("url")
    @set:JsonProperty("url")
    @JsonProperty("url")
    var url: String? = null

    @get:JsonProperty("image")
    @set:JsonProperty("image")
    @JsonProperty("image")
    var image: Any? = null

    @get:JsonProperty("createdby")
    @set:JsonProperty("createdby")
    @JsonProperty("createdby")
    var createdby: String? = null

    @get:JsonProperty("altlink")
    @set:JsonProperty("altlink")
    @JsonProperty("altlink")
    var altlink: String? = null

    @JsonIgnore
    private val additionalProperties: MutableMap<String, Any> = HashMap()
    @JsonAnyGetter
    fun getAdditionalProperties(): Map<String, Any> {
        return additionalProperties
    }

    @JsonAnySetter
    fun setAdditionalProperty(name: String, value: Any) {
        additionalProperties[name] = value
    }

    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Author) return false
        val author = other
        return id == author.id &&
                firstname == author.firstname &&
                lastname == author.lastname &&
                dob == author.dob &&
                profile == author.profile &&
                contact == author.contact &&
                address == author.address &&
                url == author.url &&
                image == author.image &&
                createdby == author.createdby &&
                altlink == author.altlink &&
                additionalProperties == author.additionalProperties
    }

    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
    override fun hashCode(): Int {
        return Objects.hash(
            id,
            firstname,
            lastname,
            dob,
            profile,
            contact,
            address,
            url,
            image,
            createdby,
            altlink,
            additionalProperties
        )
    }
}